README.rst

```diff
@@ -36,6 +36,44 @@ administrator must populate initially.
 #. Log in to the headnode, set up a PXE server, reboot the nodes, and deploy an
    operating system on them via the network.
 
+Requirements
+============
+
+Required software/hardware for running a production HaaS include:
+
+* Network switches:
+
+  * At least one switch from the Cisco Nexus 5xxx or Dell PowerConnect 55xx families
+  * For environments including more than one switch, all VLANs must be trunked to all managed switches
+
+* A single node that has the following:
+
+  * A webserver capable of supporting the WSGI standard (Apache/mod_wsgi is the only one tested)
+  * python 2.7, with the ability to install packages via pip
+  * Access to:
+
+    * The Internet or intranet (a way for users to connect to the HaaS service)
+    * The administrative telnet IP on the managed switches
+
+  * Currently only CentOS and RHEL 7.x have been tested, though any node that otherwise meets these requirements should function.
+
+* Database: a Postgres database server. Sqlite works but is not recommended for production.
+
+Optional components
```

Comment:
re: headnodes, I actually think you could get away with just not definining any base images. The api calls would still be there, but there would be no way for a user to actually make a headnode. Still, it's a bit of a hack.

I'd like to just not have any of this tagged as optional for now. We'll update it when @SahilTikale's OBM stuff is in, and we can do so again when we rip out the headnodes.

I think the null OBM driver is a good idea; We seem to be gradually replacing `@no_dry_run` with that sort of thing anyway. We'll also have the recursive OBM driver at some point.